python/lsst/daf/butler/datastore/stored_file_info.py

```diff
@@ -87,13 +89,14 @@ def from_record(cls: type[StoredDatastoreItemInfo], record: Mapping[str, Any]) -
         """
         raise NotImplementedError()
 
-    def to_record(self) -> dict[str, Any]:
-        """Convert record contents to a dictionary."""
-        raise NotImplementedError()
+    def to_record(self, **kwargs: Any) -> dict[str, Any]:
```

Comment:
This may not be time to change it, but I'm not a fan of the name of this; we call these things "records" before we convert them to `dict`, so there's nothing here that implies conversion to `dict`.

Reply:
Yes, the naming is confusing, we call instances infos or records in different places. I can rename this method (and `from_record`, `to_records`, `from_records`) to `as_dict`, etc. @timj, do you have preference for keeping this naming or some other names?

Reply:
Renaming `StoredDatastoreItemInfo` -> `DatastoreRecord`  and `StoredFileInfo` -> `FileDatastoreRecord` would be ideal but maybe not a good idea (especially on this ticket).

I do think `to_dict` is better than `to_records` (same with `from_`), but I think we may well want these to be native pydantic models, which would mean we'd just call `model_dump` instead of having another method.

Reply:
OK, I'll leave these as they are for now, and wait for big refactoring.